lib/workflow/base.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+module Workflow
+  class Base
```

Comment:
Realize it's a work in progress but the comments below and some of docs/workflows.md could become yard docs with examples inline.